tests/console/ovs_client.pm

```diff
@@ -0,0 +1,126 @@
+#SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+#
+# Summary: Test IPsec tunnel in Open vSwitch with 3 different authentication methods
+#
+#   This test does the following
+#    - Installs openvswitch-ipsec and openvswitch-pki
+#    - Starts the systemd service unit
+#    - Executes IPsec tunneling between two hosts with the following authenitcation
+#    methods:
+#       * Pre-shared key
+#       * Self-signed certificate
+#       * CA-signed certificate
+#
+# Maintainer: Anna Minou <anminou@suse.de>
+#
+use base "consoletest";
+use strict;
+use warnings;
+use testapi;
+use lockapi;
+use version_utils;
+use utils;
+use console::ovs_utils;
+
+
+my $server_ip   = "10.0.2.101";
+my $client_ip   = "10.0.2.102";
+my $id          = "2";
+my $dir         = "/etc/keys/";
+my $dir_certs   = "/etc/ipsec.d/certs/";
+my $dir_private = "/etc/ipsec.d/private/";
+my $dir_cacerts = "/etc/ipsec.d/cacerts/";
+
+sub run {
+
+    my ($self) = @_;
+    $self->select_serial_terminal;
+
+    # Install the needed packages
+    zypper_call('in openvswitch-ipsec openvswitch-pki', timeout => 300);
+
+    # Start the openvswitch and openvswitch-ipsec services
+    systemctl 'start openvswitch',       timeout => 200;
+    systemctl 'start openvswitch-ipsec', timeout => 200;
+
+    # Setup ovs bridge
+    add_bridge("$id");
+
+    # Set IPsec tunnel using pre-shared key
+    assert_script_run("ovs-vsctl add-port br-ipsec tun -- set interface tun type=gre options:remote_ip=$server_ip options:psk=swordfish");
+
+    # Wait till both hosts have finished the IPsec setup
+    barrier_wait 'ipsec_done';
+
+    # Wait for the server to finish checking the sent ESP packets
+    barrier_wait 'traffic_check_done';
+
+
+    assert_script_run("ovs-vsctl del-br br-ipsec");
+    add_bridge("$id");
+    assert_script_run("mkdir $dir && cd $dir");
+
+    # Set IPsec tunnel using self-signed certificate
+    # Generate self-signed certificate
+    assert_script_run("ovs-pki req -u host_2");
+    assert_script_run("ovs-pki self-sign host_2");
+
+    # Copy the certificate to server
+    assert_script_run('ssh-keygen -b 2048 -t rsa -q -N "" -f ~/.ssh/id_rsa');
+    exec_and_insert_password("ssh-copy-id -o StrictHostKeyChecking=no root\@$server_ip");
+    barrier_wait 'password_ok';
```

Comment:
I think you don't need to wait for the server here.

Reply:
yes, it's true, another barrier has to be added two lines upper, to make sure that certificate is ready to be copied over ssh in order to avoid this : http://10.163.41.158/tests/591#step/ovs_server/86

Reply:
I see. Good catch ;)